src/bundle.ts

```diff
@@ -0,0 +1,87 @@
+import * as path from "path";
+import { SourceNode } from "source-map";
+import * as ts from "typescript";
+import { couldNotFindBundleEntryPoint } from "./diagnostics";
+import { resolveFromRootDir } from "./resolve";
+import { EmitHost, TranspiledFile } from "./Transpile";
+import { formatPathToLuaPath, trimExtension } from "./utils";
+
+const formatPath = (path: string) => formatPathToLuaPath(trimExtension(path));
+
+export function bundleTranspiledFiles(
+    bundleFile: string,
+    entryModule: string,
+    transpiledFiles: TranspiledFile[],
+    program: ts.Program,
+    emitHost: EmitHost
+): [ts.Diagnostic[], TranspiledFile] {
+    const diagnostics: ts.Diagnostic[] = [];
+
+    const resolvedEntryModule = resolveFromRootDir(program, entryModule);
+    if (!transpiledFiles.some(f => resolveFromRootDir(program, f.fileName) === resolvedEntryModule)) {
+        return [[couldNotFindBundleEntryPoint(entryModule)], { fileName: bundleFile }];
+    }
+
+    // For each file: ["<file name>"] = function() <lua content> end,
+    const projectRootDir = program.getCommonSourceDirectory();
+    const moduleTableEntries: SourceChunk[] = transpiledFiles.map(f => moduleSourceNode(f, projectRootDir));
+
+    // If any of the modules contains a require for lualib_bundle, add it to the module table.
+    const lualibRequired = transpiledFiles.some(f => f.lua && f.lua.match(/require\("lualib_bundle"\)/));
+    if (lualibRequired) {
+        const lualibBundle = emitHost.readFile(path.resolve(__dirname, "../dist/lualib/lualib_bundle.lua"));
+        moduleTableEntries.push(`["lualib_bundle"] = function() ${lualibBundle} end,\n`);
+    }
+
+    // Create ____modules table containing all entries from moduleTableEntries
+    const moduleTable = createModuleTableNode(moduleTableEntries);
+
+    // Override `require` to read from ____modules table.
+    const requireOverride =
+        `local ____moduleCache = {}\n` +
+        `local ____originalRequire = require\n` +
+        `function require(file) if ____moduleCache[file] then return ____moduleCache[file] end\n` +
+        `if ____modules[file] then ____moduleCache[file] = ____modules[file](); return ____moduleCache[file] ` +
+        `else print("Could not find module '"..file.."' to require."); return ____originalRequire(file) end end\n`;
+    const entryPoint = `return require("${formatPath(entryModule)}")\n`;
```

Comment:
```suggestion
    const entryPoint = `return require("${escapeString(formatPath(entryModule))}")\n`;
```

Reply:
Also, it should use the same logic as one used in `moduleSourceNode`, with `resolvedEntryModule`